src/types/type_param.rs

```diff
@@ -6,31 +6,26 @@
 
 use thiserror::Error;
 
-use crate::ops::constant::typecheck::{check_int_fits_in_width, ConstIntError};
-use crate::ops::constant::HugrIntValueStore;
+use crate::ops::constant::{CustomConst, CustomSerialized};
 
-use super::{simple::Container, ClassicType, HashableType, PrimType, SimpleType, TypeTag};
+use super::CustomType;
+use super::{ClassicType, PrimType, SimpleType, TypeTag};
 
 /// A parameter declared by an OpDef. Specifies a value
 /// that must be provided by each operation node.
-// TODO any other 'leaf' types? We specifically do not want float.
-// bool should eventually be a Sum type (Container).
 #[derive(Clone, Debug, PartialEq, Eq, serde::Deserialize, serde::Serialize)]
 #[non_exhaustive]
 pub enum TypeParam {
-    /// Argument is a [TypeArg::Type] - classic or linear
-    Type,
-    /// Argument is a [TypeArg::ClassicType] - hashable or otherwise
-    ClassicType,
-    /// Argument is a [TypeArg::HashableType]
-    HashableType,
-    /// Node must provide a [TypeArg::List] (of whatever length)
-    /// TODO it'd be better to use [`Container`] here.
-    ///
-    /// [`Container`]: crate::types::simple::Container
+    /// A type as parameter, with given type tag.
+    Type(TypeTag),
+    /// 64-bit unsigned integer
+    Usize,
```

Comment:
Ok, this'll be the killer....`Usize` or `USize`? I feel that "usize" is not one word; it's an abbreviation for "unsigned size" and as such should be "USize". Or "Size"...